.github/workflows/release.yml

```diff
@@ -1,96 +0,0 @@
-name: Release
-on:
-  push:
-    tags:
-      - "v[0-9]+.[0-9]+.[0-9]+"
-      - "v[0-9]+.[0-9]+.[0-9]+rc[0-9]+"
-      - "v[0-9]+.[0-9]+.[0-9]+dev[0-9]+"
-  workflow_dispatch:
-    inputs:
-      skip-tests:
-        type: boolean
-        description: Skip
-        default: false
-      ref:
-        type: string
-        description: Tag to release
-        required: true
-
-permissions:
-  contents: read
-
-jobs:
-  pre-commit:
-    uses: ./.github/workflows/step_pre-commit.yml
-    if: "! inputs.skip-tests"
-
-  test-pip:
-    needs: [ pre-commit ]
-    uses: ./.github/workflows/step_tests-pip.yml
-    with:
-      coverage: false
-    if: "! inputs.skip-tests"
-
-  test-conda:
-    needs: [ test-pip ]
-    uses: ./.github/workflows/step_tests-conda.yml
-    with:
-      coverage: false
-    if: "! inputs.skip-tests"
-
-  test-ui:
-    needs: [ test-pip ]
-    uses: ./.github/workflows/step_tests-ui.yml
-    if: "! inputs.skip-tests"
-
-  test-status:
-    needs: [ pre-commit, test-pip, test-conda, test-ui ]
-    runs-on: ubuntu-latest
-    steps:
-      - uses: re-actors/alls-green@release/v1
-        with:
-          allowed-skips: pre-commit, test-pip, test-conda, test-ui
-          jobs: ${{ toJSON(needs) }}
-    if: always()
-
-  build:
-    needs: [ test-status ]
-    uses: ./.github/workflows/step_build.yml
-    with:
-      upload: true
-      ref: ${{ inputs.ref }}
```

Comment:
This one is the same workflow as you have currently, it's just separated into steps to minimize maintenance and to scope the permissions a bit